testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -1455,6 +1462,134 @@ public void testTrinoReadsSparkRowLevelDeletesWithRowTypes(StorageFormat tableSt
         onSpark().executeQuery("DROP TABLE " + sparkTableName);
     }
 
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS}, dataProvider = "storageFormats")
+    public void testSparkReadsTrinoRowLevelDeletes(StorageFormat storageFormat)
+    {
+        String tableName = format("test_spark_reads_trino_row_level_deletes_%s_%s", storageFormat.name(), randomTableSuffix());
+        String sparkTableName = sparkTableName(tableName);
+        String trinoTableName = trinoTableName(tableName);
+
+        onTrino().executeQuery("CREATE TABLE " + trinoTableName + "(a INT, b INT) WITH(partitioning = ARRAY['b'], format_version = 2, format = '" + storageFormat.name() + "')");
+        onTrino().executeQuery("INSERT INTO " + trinoTableName + " VALUES (1, 2), (2, 2), (3, 2), (11, 12), (12, 12), (13, 12)");
+        // Delete one row in a file
+        onTrino().executeQuery("DELETE FROM " + trinoTableName + " WHERE a = 13");
+        // Delete an entire partition
+        onTrino().executeQuery("DELETE FROM " + trinoTableName + " WHERE b = 2");
+
+        List<Row> expected = ImmutableList.of(row(11, 12), row(12, 12));
+        assertThat(onTrino().executeQuery("SELECT * FROM " + trinoTableName)).containsOnly(expected);
+        assertThat(onSpark().executeQuery("SELECT * FROM " + sparkTableName)).containsOnly(expected);
+
+        // Delete to a file that already has deleted rows
+        onTrino().executeQuery("DELETE FROM " + trinoTableName + " WHERE a = 12");
+        expected = ImmutableList.of(row(11, 12));
+        assertThat(onTrino().executeQuery("SELECT * FROM " + trinoTableName)).containsOnly(expected);
+        // TODO: Uncomment this after disabling Spark caching
```

Comment:
https://github.com/trinodb/trino/pull/11965

Reply:
@alexjo2144 if this is more feasible for you to disable spark caching here go for it. I am sorry if I messed with your pr I was not aware

Reply:
Not at all, I was just leaving a reminder for myself. I think your pr will certainly be merged before this one

Reply:
https://github.com/trinodb/trino/pull/11965 is merged

Reply:
applied myself